src/main/java/org/carlmontrobotics/robotcode2023/subsystems/Roller.java

```diff
@@ -89,6 +93,41 @@ public double getGamePieceDistanceIn() {
                                                                                 */) / 1000 /* Convert mm to m */);
     }
 
+    // This assumes robot's middle is exactly in line with the poles where you place
+    // the cones
+    // outtake side is the same side as the limelight
+    public Pose2d correctPosition() {
+        Pose2d initialPose = dt.getPose();
+        double distanceToMove = 0;
+        double dist = getGamePieceDistanceIn();
+        Translation2d translation = new Translation2d(0, 0);
+        if (dist < LEFT_LIMIT) {
+            // need to shift robot right
+            distanceToMove = ROLLER_WIDTH / 2 - dist;
+            translation = new Translation2d(distanceToMove,
+                    initialPose.getRotation().plus(new Rotation2d(Math.PI / 2)));
+
+        } else if (dist > RIGHT_LIMIT) {
+            // need to shift robot left
+            distanceToMove = dist - ROLLER_WIDTH / 2;
+            translation = new Translation2d(distanceToMove,
+                    initialPose.getRotation().plus(new Rotation2d(-Math.PI / 2)));
+        }
+        Transform2d transform = new Transform2d(translation, new Rotation2d(0));
+        return initialPose.plus(transform);
+    }
```

Comment:
This method will probably be rewritten/replaced when you address other comments, but a few notes:
1. You don't seem to account for the width of the game piece.
2. You are always rotating relative to the initial pose, which doesn't sound right.